test/NuGet.Clients.Tests/NuGet.CommandLine.Test/ExtractOptionsTests.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class ExtractOptionsTests
+    {
+
+        [Fact]
+        public void ExtractOption_PassOptionNoHttpCache_NoHttpCacheShouldBeSet()
+        {
+            //setup
+            RestoreCommand command = new RestoreCommand();
+            List<string> args = new() { "-NoHttpCache" };
+            ICommandManager manager = new CommandManager();
+            CommandLineParser commandLineParser = new CommandLineParser(manager);
+
+            //execute extract options
+            commandLineParser.ExtractOptions(command, args.GetEnumerator());
+
+            //assert
+            Assert.True(command.NoHttpCache);
+        }
+
+        [Fact]
+        public void ExtractOption_DoNotPassOptionNoHttpCache_NoHttpCacheShouldNotBeSet()
```

Comment:
This test currently tests that RestoreCommand's default options are to set NoCache to false. However, the test name suggests that not passing an option to ExtractOption, will result in NoHttpCache being not set. Furthermore, we are testing ExtractOption and we cannot guarantee that the RestoreCommand default behavior will never change so we need to be explicit about how that is set up.

Let's change the name to reflect the fact we're testing ExtractOption with a command that has NoCache set to false, results in the NoHttpCache set to false.